server/remote_cache/byte_stream_server/byte_stream_server.go

```diff
@@ -342,3 +380,107 @@ func (s *ByteStreamServer) QueryWriteStatus(ctx context.Context, req *bspb.Query
 		Complete:      false,
 	}, nil
 }
+
+type Checksum struct {
```

Comment:
nit: see above but if we simplify the above -- do we need this class?

Reply:
The decompressor encloses a Writer so I needed this class to satisfy the Writer interface. I figured it made sense to move the other checksum related stuff in here as well.